### PR TITLE
merge unity-fork, part 2

### PR DESCRIPTION
**Pull request by piqiums**
merge unity-fork, part 2
Also addressed all Ashish's comments in https://github.com/AzureAD/azure-activedirectory-library-for-android/pull/1145, which is obsoleted since tracking a wrong branch.

### CODE REVIEW

File: adal/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -11,7 +11,6 @@
     <application>
         <activity android:name="com.microsoft.aad.adal.AuthenticationActivity" >
         </activity>
-
```
Comment (addev-ashish) [inline adal/src/main/AndroidManifest.xml]: nit: remove this file from PR

File: adal/src/main/java/com/microsoft/aad/adal/AuthenticationConstants.java [hunk 2]
```diff
@@ -108,6 +111,7 @@ private AuthenticationConstants() { }
          * Waiting for broker package to be installed, and resume request in broker. 
          */
         public static final int BROKER_REQUEST_RESUME = 2006;
+        public static final int BROWSER_CODE_DEVICE_REGISTER = 2006;
```
Comment (addev-ashish) [inline adal/src/main/java/com/microsoft/aad/adal/AuthenticationConstants.java]: you should not be using the same number twice, see where this number is coming from

File: adal/src/main/java/com/microsoft/aad/adal/AuthenticationConstants.java [hunk 3]
```diff
@@ -477,6 +526,7 @@ private AuthenticationConstants() { }
 
         /** The value for pkeyauth redirect. */
         public static final String PKEYAUTH_REDIRECT = "urn:http-auth:PKeyAuth";
+        public static final String CLIENT_TLS_REDIRECT = "urn:http-auth:PKeyAuth";
```
Comment (addev-ashish) [inline adal/src/main/java/com/microsoft/aad/adal/AuthenticationConstants.java]: exactly same as previous line

File: adal/src/main/java/com/microsoft/aad/adal/PackageHelper.java [hunk 4]
```diff
@@ -58,6 +62,43 @@
...
+    /**
+     * Gets metadata information from AndroidManifest file.
+     *
+     * @param packageName
+     * @param component
+     * @param metaDataName
+     * @return MetaData
+     */
+    @SuppressLint("WrongConstant")
+    Object getValueFromMetaData(final String packageName, final ComponentName component,
```
Comment (addev-ashish) [inline adal/src/main/java/com/microsoft/aad/adal/PackageHelper.java]: where is this used?

File: adal/src/main/AndroidManifest.xml (hunk 1 above)
Comment (piqiums) [reply adal/src/main/AndroidManifest.xml]: this is removing the redundant empty line in last PR as you commented.

File: adal/src/main/java/com/microsoft/aad/adal/AuthenticationConstants.java (hunk 3 above)
Comment (piqiums) [reply adal/src/main/java/com/microsoft/aad/adal/AuthenticationConstants.java]: CLIENT_TLS_REDIRECT used by other packages. I would like to merge it in this way first, and do a small refactor later to update all files using this variable name. Does this make sense ?

File: adal/src/main/java/com/microsoft/aad/adal/AuthenticationConstants.java (hunk 2 above)
Comment (addev-ashish) [reply adal/src/main/java/com/microsoft/aad/adal/AuthenticationConstants.java]: File one issue for both.

File: adal/src/main/java/com/microsoft/aad/adal/PackageHelper.java (hunk 4 above)
Comment (addev-ashish) [reply adal/src/main/java/com/microsoft/aad/adal/PackageHelper.java]: Used in a test case

**Pull request merged**